homework-submission/react-week2/src/App.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import HyfButton from "./components/HyfButton";
+import HyfInput from "./components/HyfInput";
+import TodoList from './components/TodoList'
+import TodoItems from './components/TodoItems'
+import "./App.css";
+
+
+function App() {
+  const uniqid = require('uniqid')
+
+  const [toDoItems, setTodoItems] = useState([
+    {
+      name: "eat cheese"
+    },
+    {
+      name: "fight monsters"
+    },
+    {
+      name: "dont fall over"
+    }
+  ]);
+
+  const [isDone, setIsDone] = useState([])
+  
+  const [addItem, setAddItem] = useState({})
+
+  
+  const handleRemove = (name) => {
+    const removeTodo = toDoItems.filter(item => item.name !== name)
+  setTodoItems(removeTodo)
+  }
+
+  const onHandleInputChange = e => {
+   const name= e.target.value
+   const newItem = { name }
+   setAddItem(newItem)
+  }
+
+  const onHandleCheckboxChange = (name) => {
+  const isCompletedItem = isDone.find(i => i.name === name)
+  if(isCompletedItem){
+    const completedItem = isDone.filter(i => i.name !== name)
+    setIsDone(completedItem)
+  }else{
+    const completedItem = toDoItems.find(i => i.name === name)
+    const completedItems = [...isDone, completedItem]
+    setIsDone(completedItems)
+  }
+  }
+
+  const onHandleClick = () => {
+    const newList = [...toDoItems, addItem]
+    setTodoItems(newList)
+  }
+
+  return <div className="App">
+  <TodoList 
+  input={<HyfInput onHandleInputChange={onHandleInputChange}/>} 
```

Comment:
Why are you passing components to the props of `TodoList`? Here you can pass just the function `onHandleInputChange` and let the `TodoList` render it's own input field. For `children` counts the same, no need to already pass rendered components, better to pass the array of values and let the child component render it's own children.